test_pull_requests

```diff
@@ -1184,8 +1304,56 @@ popd
         # comment to reflect the new state
         if submit_test_job
           delete_comment_with_prefix(id, base_repo, ACTION_PREFIX, comments)
-          extended_tests = get_extended_tests(req, comments, branch, settings)
-          submit_tests(repo_to_pull_request, base_repo, branch, comment_id, extended_tests, settings)
+          # Check for pretest_settings_key, so we might skip a round
+          # of tests prior to merge
+          preceding_test_settings = Properties['settings'][settings['pretest_settings_key']]
+          test_build_url = ''
+          quick_merge_comments = []
+          can_skip_merge_tests = false
+          if !preceding_test_settings.nil?
+            p_test_prefix = preceding_test_settings['test_prefix']
+            coreq_tests_synced_and_passed = repo_to_pull_request.all? do |repo, pull_request|
+              preceding_test_comment = get_comment_with_prefix(pull_request['number'], repo, p_test_prefix, (repo == base_repo ? comments : nil))
+              if preceding_test_comment.nil?
+                false
+              else
+                p_comment, p_test_build_url, p_base_commit, p_extended_tests = extract_bot_comment_metadata(preceding_test_comment['body'], preceding_test_settings)
+                p_base_commit = base_commit_tag_to_sha(p_base_commit)
+                can_skip_one = (p_comment =~ /^SUCCESS$/ &&
+                                p_base_commit == pull_request['base']['sha'])
+                if can_skip_one
+                  # Store pre-merge comments so we can carry the build url and base commit id through to SUCCESS
+                  quick_merge_comments << [pull_request['number'], repo, compose_bot_comment(settings['test_prefix'], 'Waiting:', :build_url => p_test_build_url, :base_commit => p_base_commit)]
+                end
+                can_skip_one
+              end
+            end
+            can_skip_merge_tests = coreq_tests_synced_and_passed
+          end
+          if can_skip_merge_tests
+            begin
+              quick_merge_comments.each do |pull_id, repo, new_comment|
+                process_or_create_comment(pull_id, repo, settings) do |c_id, c_comment, c_updated_at|
+                  update_comment(c_id, repo, new_comment)
+                end
+              end
+              $stderr.puts "Skipping #{settings['name']} tests for pull request ##{id} for repo '#{base_repo}'"
+              repo_to_pull_request.each do |repo, pull_request|
+                test_merge_pull_request(pull_request['number'], repo, settings)
+              end if repo_to_pull_request.length > 0
+              repo_to_pull_request.each do |repo, pull_request|
+                $stderr.puts "\n*******Merging tested, up-to-date pull request: #{GITHUB_BASE_URL}/#{Properties['github_user']}/#{repo}/pull/#{pull_request['number']} "
+                merge_pull_request(pull_request['number'], repo, settings)
+              end
+            rescue Exception => e
```

Comment:
You're right, it's silly. I'll remove it when I get a chance to work on this some more tomorrow